tests/h/views/client_test.py

```diff
@@ -19,6 +22,80 @@ def test_annotator_token_returns_token(generate_jwt, pyramid_request):
     assert result == generate_jwt.return_value
 
 
+@pytest.mark.usefixtures('routes')
+class TestSidebarApp(object):
+
+    def test_it_includes_client_config(self, pyramid_request_):
+        ctx = client.sidebar_app(pyramid_request_)
+        expected_config = {
+                'apiUrl': 'http://example.com/api',
+                'websocketUrl': 'wss://example.com/ws',
+                'serviceUrl': 'http://example.com/',
+                'release': __version__,
+                'raven': {
+                    'dsn': 'test-sentry-dsn',
+                    'release': __version__
+                },
+                'googleAnalytics': 'UA-4567'
+                }
+        actual_config = json.loads(ctx['app_config'])
+        assert actual_config == expected_config
+
+    def test_it_sets_asset_urls(self, pyramid_request_):
+        ctx = client.sidebar_app(pyramid_request_)
+        assert ctx['app_css_urls'] == ['/assets/client/app.css']
+        assert ctx['app_js_urls'] == ['/assets/client/app.js']
+
+
+@pytest.mark.usefixtures('routes')
+class TestEmbed(object):
+    def test_it_sets_sidebar_app_url(self, pyramid_request_):
+        ctx = client.embed(pyramid_request_)
+        assert ctx['app_html_url'] == 'http://example.com/app.html'
+
+    def test_it_sets_asset_urls(self, pyramid_request_):
+        ctx = client.embed(pyramid_request_)
+        assert ctx['inject_resource_urls'] == [
+            'http://example.com/assets/client/polyfills.js',
+            'http://example.com/assets/client/inject.js',
+            'http://example.com/assets/client/inject.css',
+            ]
+
+    def test_it_sets_content_type(self, pyramid_request_):
+        client.embed(pyramid_request_)
+        assert pyramid_request_.response.content_type == 'text/javascript'
+
+
+@pytest.fixture
+def pyramid_request_(pyramid_request):
```

Comment:
Good question - we had a discussion on Slack and Nick suggested a cleaner solution which I am now implementing. See https://hypothes-is.slack.com/archives/public/p1486725096004540